sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -586,20 +638,39 @@ def _fit_stochastic(
             # don't stratify in multilabel classification
             should_stratify = is_classifier(self) and self.n_outputs_ == 1
             stratify = y if should_stratify else None
-            X, X_val, y, y_val = train_test_split(
-                X,
-                y,
-                random_state=self._random_state,
-                test_size=self.validation_fraction,
-                stratify=stratify,
-            )
+            if sample_weight is None:
+                X_train, X_val, y_train, y_val = train_test_split(
+                    X,
+                    y,
+                    random_state=self._random_state,
+                    test_size=self.validation_fraction,
+                    stratify=stratify,
+                )
+                sample_weight_train = sample_weight_val = None
+            else:
+                # TODO: incorporate sample_weight in sampling here.
+                (
```

Comment:
It could be cool to cover this part as well by activating `early_stopping` together with `sample_weight`.

With the work of @adrinjalali, we can easily covered it in the common test by setting the parameter value with 

```python
    MLPRegressor: {
        "check_sample_weight_equivalence": [
            dict(solver="sgd", tol=1e-2, random_state=42),
            dict(early_stopping=True, solver="sgd", tol=1e-2, random_state=42),
        ]
    },
```

in `PER_ESTIMATOR_CHECK_PARAMS`. We can do the same for the classifier.

Reply:
I won't do it in this PR. This PR steals this part from HGBT. If we fix it, we can fix it  for both .... in another PR - if ever.

Reply:
I'm not sure to understand:

```python
    MLPClassifier: {
        "check_sample_weight_equivalence": [
            dict(solver="lbfgs"),
            dict(solver="lbfgs", early_stopping=True, n_iter_no_change=1, tol=1e-2),
        ]
    },
    MLPRegressor: {
        "check_sample_weight_equivalence": [
            dict(solver="sgd", tol=1e-2, random_state=42),
            dict(solver="lbfgs", early_stopping=True, n_iter_no_change=1, tol=1e-2),
        ]
    },
```

This is passing the test of the equivalence, I don't ask to to change anything else.

In HGBDT, we have a test that check that when `early_stopping` is activated, we are passing the weights to the scoring function.

Reply:
Is [d53090a](https://github.com/scikit-learn/scikit-learn/pull/30155/commits/d53090a878ef8594709107c358fd9ab5b369310e) enough?

Reply:
It seems like this code branch is still not covered. We may need to set the `early_stopping` parameter to True to cover this part.